daq/network.py

```diff
@@ -72,7 +72,10 @@ def __init__(self, config):
         self._used_ip_indices = set()
         self.topology = FaucetTopology(self.config)
         switch_setup = config.get('switch_setup', {})
-        self.ext_intf = switch_setup.get('data_intf')
+        if self.config.get('run_trigger', {}).get('auto_session'):
```

Comment:
Not a big fan of having this logic duplicated. I think it would be trivial to have network.py pull it from topology... i.e.

`self.ext_intf = self.topology.ext_intf`

and then leave the if statement logic in just that one place.

Reply:
Changed.